File: Assets/Scripts/GirlHeadMainScript.cs
using System.Collections;
using UnityEngine;

[System.Serializable]
public partial class GirlHeadMainScript : MonoBehaviour
{
    public GameObject Tear;
    public Vector3[] positions;
    public int[] indexes;
    public int index;
    public virtual void Start()
    {
        InvokeRepeating("CreateTear", 2, 1);
        RandomizeIndexes();
        positions = new Vector3[8];
        positions[0] = new Vector3(0.1438493f, 0.6631676f, -0.1060616f);
        positions[1] = new Vector3(0.2244534f, 0.6696064f, -0.1060616f);
        positions[2] = new Vector3(0.07132857f, 0.624287f, -0.1060616f);
        positions[3] = new Vector3(0.2944834f, 0.7581771f, -0.1060616f);
        positions[4] = new Vector3(-0.1521176f, 0.4364915f, -0.1060616f);
        positions[5] = new Vector3(-0.2318641f, 0.3938793f, -0.1060616f);
        positions[6] = new Vector3(-0.4313891f, 0.342688f, -0.1060616f);
        positions[7] = new Vector3(-0.3340824f, 0.3427447f, -0.1060616f);
    }

    public virtual void CreateTear()
    {
        index++;
        if (index == 6)
        {
            index = 0;
        }
        if (index >= 4)
        {
            return;
        }
        StartCoroutine(CreateTear2(index));
        StartCoroutine(CreateTear2(index + 4));
    }

    public virtual IEnumerator CreateTear2(int i)
    {
        yield return new WaitForSeconds(Random.Range(0, 0.3f));
        GameObject g = UnityEngine.Object.Instantiate(Tear, transform.position, transform.rotation);
        g.transform.Rotate(0, 180, 0);
        g.transform.parent = transform;
        g.transform.localPosition = positions[indexes[i]];
    }

    public virtual void RandomizeIndexes()
    {
        indexes = new int[8];
        int i = 0;
        while (i < 8)
        {
            indexes[i] = i;
            i++;
        }
        SwapTwo();
        SwapTwo();
        SwapTwo();
    }

    public virtual void SwapTwo()
    {
        int temp = 0;
        int a = Random.Range(0, 4);
        int b = Random.Range(0, 4);
        temp = indexes[a];
        indexes[a] = indexes[b];
        indexes[b] = temp;
        a = Random.Range(4, 8);
        b = Random.Range(4, 8);
        temp = indexes[a];
        indexes[a] = indexes[b];
        indexes[b] = temp;
    }

}

File: Assets/Scripts/GirlHeadMainScript.js
#pragma strict

var Tear:GameObject;
var positions:Vector3[];
var indexes:int[];
var index:int;

function Start () 
{
	InvokeRepeating("CreateTear",2,1);	
	RandomizeIndexes();
	
	positions=new Vector3[8];
	positions[0]=Vector3(0.1438493,0.6631676,-0.1060616);
	positions[1]=Vector3(0.2244534,0.6696064,-0.1060616);
	positions[2]=Vector3(0.07132857,0.624287,-0.1060616);
	positions[3]=Vector3(0.2944834,0.7581771,-0.1060616);
	positions[4]=Vector3(-0.1521176,0.4364915,-0.1060616);
	positions[5]=Vector3(-0.2318641,0.3938793,-0.1060616);
	positions[6]=Vector3(-0.4313891,0.342688,-0.1060616);
	positions[7]=Vector3(-0.3340824,0.3427447,-0.1060616);
}

function CreateTear()
{

	index++;
	if(index==6)
		index=0;
	if (index>=4)
		return;
		
	CreateTear2(index);
	CreateTear2(index+4);

}

function CreateTear2( i:int)
{
	yield WaitForSeconds(Random.Range(0,.3));
	var g:GameObject=Instantiate(Tear,transform.position,transform.rotation);
	g.transform.Rotate(0,180,0);
	g.transform.parent=transform;
	g.transform.localPosition=positions[indexes[i]];

}

function RandomizeIndexes()
{
	indexes=new int[8];

	for (var i:int=0;i<8;i++)
	{
		indexes[i]=i;
	}
	
	SwapTwo();
	SwapTwo();
	SwapTwo();
}


function SwapTwo()
{
	var a:int=Random.Range(0,4);
	var b:int=Random.Range(0,4);
	var temp:int;
	
	
	temp=indexes[a];
	indexes[a]=indexes[b];
	indexes[b]=temp;


	a=Random.Range(4,8);
	b=Random.Range(4,8);
	
	
	temp=indexes[a];
	indexes[a]=indexes[b];
	indexes[b]=temp;

}










